pkg/pushsync/pushsync.go

```diff
@@ -183,78 +186,55 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 
 	price := ps.pricer.Price(chunkAddress)
 
-	// if the peer is closer to the chunk, AND it's a full node, we were selected for replication. Return early.
-	if p.FullNode {
-		if closer, _ := p.Address.Closer(chunkAddress, ps.address); closer {
-
-			ps.metrics.HandlerReplication.Inc()
+	if ch.Replicate {
 
-			var err error
-			defer func() {
-				if err != nil {
-					ps.metrics.HandlerReplicationErrors.Inc()
-				}
-			}()
+		ps.metrics.HandlerReplication.Inc()
 
-			ctxd, canceld := context.WithTimeout(context.Background(), replicationTTL)
-			defer canceld()
-
-			span, _, ctxd := ps.tracer.StartSpanFromContext(ctxd, "pushsync-replication-storage", ps.logger, opentracing.Tag{Key: "address", Value: chunkAddress.String()})
-			defer span.Finish()
-
-			chunk, err = ps.validStamp(chunk, ch.Stamp)
+		var err error
+		defer func() {
 			if err != nil {
-				return fmt.Errorf("pushsync replication invalid stamp: %w", err)
+				ps.metrics.HandlerReplicationErrors.Inc()
 			}
+		}()
```

Comment:
Final error handling on line 193 seems to be shadowing error handling 148. 
If `ch.Replicate` is true, would error handling on line 148 get swallowed in case of error? Is this by intention?